### PR TITLE
Fix for Guice Wrapped Class

### PR DESCRIPTION
**Pull request by vsgamb**

### CODE REVIEW

File: tef-impl/src/main/java/flipkart/tef/bizlogics/DataAdapterBizlogic.java [hunk 1]
```diff
@@ -80,6 +81,11 @@ private Map<DataAdapterKey<?>, Field> buildCacheOfMutableFields() {
 
     @SuppressWarnings("rawtypes")
     public static String getEmittedDataName(Class<? extends DataAdapterBizlogic> clazz) {
+        if (clazz.getName().contains(BytecodeGen.ENHANCER_BY_GUICE_MARKER)) {
```
Comment (bageshwar) [inline tef-impl/src/main/java/flipkart/tef/bizlogics/DataAdapterBizlogic.java:84]: Let's also mention why this change was needed in comments

File: tef-impl/src/test/java/flipkart/tef/bizlogics/DataAdapterBizlogicTest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                    new CustomInterceptor()
+            );
+        }
+    }
+
+    public class CustomInterceptor implements MethodInterceptor {
+        @Override
+        public Object invoke(MethodInvocation invocation) throws Throwable {
+            // Log method details before invocation
+            System.out.println("Before test method invocation");
```
Comment (bageshwar) [inline tef-impl/src/test/java/flipkart/tef/bizlogics/DataAdapterBizlogicTest.java:86]: Do we need this log lines?

File: tef-impl/src/test/java/flipkart/tef/bizlogics/DataAdapterBizlogicTest.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
+package flipkart.tef.bizlogics;
```
Comment (bageshwar) [inline tef-impl/src/test/java/flipkart/tef/bizlogics/DataAdapterBizlogicTest.java:1]: Add copyright header (check peer class files)

File: tef-impl/src/main/java/flipkart/tef/bizlogics/DataAdapterBizlogic.java [hunk 4]
```diff
@@ -80,6 +81,11 @@ private Map<DataAdapterKey<?>, Field> buildCacheOfMutableFields() {
 
     @SuppressWarnings("rawtypes")
     public static String getEmittedDataName(Class<? extends DataAdapterBizlogic> clazz) {
```
Comment (bageshwar) [inline tef-impl/src/main/java/flipkart/tef/bizlogics/DataAdapterBizlogic.java:83]: Can you check once if this method really has to be public? If not, can we make it private and run the test via reflection, or make it package private atleast.

**Review by bageshwar — CHANGES_REQUESTED** [review]

File: tef-impl/src/main/java/flipkart/tef/bizlogics/DataAdapterBizlogic.java (hunk 4 above)
Comment (vsgamb) [reply tef-impl/src/main/java/flipkart/tef/bizlogics/DataAdapterBizlogic.java:83]: Its been used in FlowBuilder also.

File: tef-impl/src/main/java/flipkart/tef/bizlogics/DataAdapterBizlogic.java (hunk 1 above)
Comment (vsgamb) [reply tef-impl/src/main/java/flipkart/tef/bizlogics/DataAdapterBizlogic.java:86]: Added the comment

File: tef-impl/src/test/java/flipkart/tef/bizlogics/DataAdapterBizlogicTest.java (hunk 3 above)
Comment (vsgamb) [reply tef-impl/src/test/java/flipkart/tef/bizlogics/DataAdapterBizlogicTest.java:17]: Added

**Review by bageshwar — APPROVED** [review]
LGTM
